### PR TITLE
Swift Playground

### PR DESCRIPTION
**Pull request by RuiAAPeres**
**What's in this Pull Request**

The goal of this PR is to make it as simple as possible for new comers to experiment with ReactiveCocoa. In it, they will find multiple examples of how different pieces work (`Signal`, `SignalProducer`, `Observer`, `Scheduler`) and of course the operators (`map`, `filter`, et all). Of course this can also  be quite valuable for a quick brainstorm or testing new operators, for more advanced users.

I think this is quite important to experiment in a highly interactive place like a playground, to start to have a feeling about a signal's behaviour in the ReactiveCocoa's context. 

**Tasks**

- [x] `SignalProducer` 
- [ ] `Signal`
- [ ] `Schedulers`
- [ ] `Properties`

I am quite happy with more ideas for what to include in it. I will also have a look at what they did on the RxSwift playground. 

Finally, this ties to #2129.

### CODE REVIEW

File: ReactiveCocoa.playground/Pages/SignalProducer.xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+/*:
+### `empty`
+A producer for a Signal that will immediately complete without sending
+any values.
+*/
+
+scopedExample("`empty`") {
+    
+    let emptyProducer = SignalProducer<Int, NoError>.empty
+    let observer = Observer<Int, NoError>(failed: { _ in print("error not called") }, completed: { print("completed called")}, next: { _ in print("next not called")})
```
Comment (mdiep) [inline ReactiveCocoa.playground/Pages/SignalProducer.xcplaygroundpage/Contents.swift]: These might benefit from formatting each block on a separate line.

**Comment by mdiep** [comment]
I like it. 👍 Do you intend to tackle the "Next Steps" here or in a separate PR?

**Comment by RuiAAPeres** [comment]
@mdiep here. I just wanted to push the work I had done! I will squash it, when it's time to merge.

**Comment by RuiAAPeres** [comment]
@mdiep what do you think about merging it with SignalProducers for now? I am feeling I am taking too long with this (not a lot of time) and I am getting behind as new stuff is added.

**Comment by mdiep** [comment]
👍 👍

**Pull request merged**